### ISSUE TITLE
Bug: `no-unused-vars` raising TypeError

### ISSUE THREAD

**Issue by CNSeniorious000** [eslint/eslint#19273]
### Environment

Node version: v22.12.0
npm version: v10.9.0
Local ESLint version: v9.17.0
Global ESLint version: Not found
Operating System: win32 10.0.26100

### What parser are you using?

Other

### What did you do?

<details>
<summary>Configuration</summary>

```js
import antfu from '@antfu/eslint-config'

export default antfu({
  svelte: true,
  typescript: true,
  formatters: true,
})
```
</details>

```svelte
{#if true}
  {#each [1, 2, 3] as a}
    {a}

    <!-- eslint-disable-next-line no-unused-vars -->
    {@const _ = console.error({ a })}

  {/each}
{/if}
```

### What did you expect to happen?

no error

### What actually happened?

```
ops! Something went wrong! :(

ESLint: 9.17.0

TypeError: Cannot read properties of undefined (reading 'length')
    at handleFixes (file:///home/projects/vitejs-vite-geepgmmf/node_modules/eslint/lib/rules/no-unused-vars.js#cjs:1162:48)
    at Object.fix (file:///home/projects/vitejs-vite-geepgmmf/node_modules/eslint/lib/rules/no-unused-vars.js#cjs:1471:48)
    at normalizeFixes (file:///home/projects/vitejs-vite-geepgmmf/node_modules/eslint/lib/linter/report-translator.js#cjs:193:28)
    at eval (file:///home/projects/vitejs-vite-geepgmmf/node_modules/eslint/lib/linter/report-translator.js#cjs:223:22)
    at Array.map (<anonymous>)
    at mapSuggestions (file:///home/projects/vitejs-vite-geepgmmf/node_modules/eslint/lib/linter/report-translator.js#cjs:217:10)
    at eval (file:///home/projects/vitejs-vite-geepgmmf/node_modules/eslint/lib/linter/report-translator.js#cjs:372:55)
    at FileContext.report (file:///home/projects/vitejs-vite-geepgmmf/node_modules/eslint/lib/linter/linter.js#cjs:1062:41)
    at Program:exit (file:///home/projects/vitejs-vite-geepgmmf/node_modules/eslint/lib/rules/no-unused-vars.js#cjs:1458:33)
    at ruleErrorHandler (file:///home/projects/vitejs-vite-geepgmmf/node_modules/eslint/lib/linter/linter.js#cjs:1098:48)
```

### Link to Minimal Reproducible Example

https://stackblitz.com/edit/eslint-no-unused-vars-repro?file=package.json

### Participation

- [ ] I am willing to submit a pull request for this issue.

### Additional comments

9.16.0 works fine.

It seems this is threw from [here](https://github.com/eslint/eslint/pull/18352/files#diff-b8867bf68ff637ad5f72d604e0e920ffced618515a50a2047958904f837f83deR1161-R1163).

**Comment by fasttime** [comment]
Thanks for the report @CNSeniorious000, I can reproduce the problem.

ESLint built-in rules like `no-unused-vars` are designed - with a few exceptions - to lint JavaScript files parsed with Espree or another ESTree compliant parser. Svelte syntax is not supported by default. This is roughly covered in [this section](https://github.com/eslint/eslint?tab=readme-ov-file#what-about-experimental-features) of the readme file.

In the case of your `src/App.svelte` file I can see that the parser is producing a nonstandard `SvelteConstTag` node with a `declaration` property instead of the expected `declarations`, and that results in the rule accessing an undefined value and ultimately crashing.

![Image](https://github.com/user-attachments/assets/65525481-5435-43cf-992e-ad71773bdaba)

The rule was updated to include suggestions in v9.17.0 and that explains why the `TypeError` didn't occur in previous versions. Note that even if the rule didn't crash in ESLint v9.16.0 there is no guarantee that it would work as intended. I'd recommend reporting this problem to [`@antfu/eslint-config`](https://github.com/antfu/eslint-config), since that's where the `no-unused-vars` rule is being enabled for Svelte files.
